pkg/collector/process.go

```diff
@@ -0,0 +1,283 @@
+	antreaRegistry registry.Registry
+	// server information
+	address net.Addr
+	// maximum buffer size to read the record
+	maxBufferSize uint16
+	// chanel to receive stop information
+	stopChan chan bool
+	// packet list
+	packets []*Packet
+}
+
+type UDPCollectingProcess struct {
+	*collectingProcess
+	// list of udp workers to process message
+	workerList []*Worker
+	// udp worker pool to receive and process message
+	workerPool chan chan *bytes.Buffer
+	// number of workers
+	workerNum int
+}
+
+type TCPCollectingProcess struct {
+	*collectingProcess
+}
+
+// data struct of processed message
+type Packet struct {
+	Version      uint16
+	BufferLength uint16
+	SeqNumber    uint32
+	ObsDomainID  uint32
+	ExportTime   uint32
+	FlowSets     interface{}
+}
+
+// Shared header fields for TemplateFlowSet and DataFlowSet
+type FlowSetHeader struct {
+	// 2 for templateFlowSet
+	// 256-65535 for dataFlowSet (templateID)
+	ID     uint16
+	Length uint16
+}
+
+type TemplateFlowSet struct {
+	FlowSetHeader
+	Records []*TemplateField
+}
+
+type DataFlowSet struct {
+	FlowSetHeader
+	Records []*DataField
+}
+
+type TemplateField struct {
+	ElementID     uint16
+	ElementLength uint16
+	EnterpriseID  uint32
+}
+
+type DataField struct {
+	DataType entities.IEDataType
+	Value    bytes.Buffer
+}
+
+func InitTCPCollectingProcess(address net.Addr, maxBufferSize uint16) (*TCPCollectingProcess, error) {
+	ianaReg := registry.NewIanaRegistry()
+	antreaReg := registry.NewAntreaRegistry()
+	ianaReg.LoadRegistry()
+	antreaReg.LoadRegistry()
+	collectProc := &TCPCollectingProcess{
+		&collectingProcess{
+			templatesMap:   make(map[uint32]map[uint16][]*TemplateField),
+			templatesLock:  &sync.RWMutex{},
+			templateTTL:    0,
+			ianaRegistry:   ianaReg,
+			antreaRegistry: antreaReg,
+			address:        address,
+			maxBufferSize:  maxBufferSize,
+			stopChan:       make(chan bool),
+			packets:        make([]*Packet, 0),
+		},
+	}
+	return collectProc, nil
+}
+
+func InitUDPCollectingProcess(address net.Addr, maxBufferSize uint16, workerNum int, templateTTL uint32) (*UDPCollectingProcess, error) {
+	ianaReg := registry.NewIanaRegistry()
+	antreaReg := registry.NewAntreaRegistry()
+	ianaReg.LoadRegistry()
+	antreaReg.LoadRegistry()
+	workerList := make([]*Worker, workerNum)
+	workerPool := make(chan chan *bytes.Buffer)
+	collectProc := &UDPCollectingProcess{
+		collectingProcess: &collectingProcess{
+			templatesMap:   make(map[uint32]map[uint16][]*TemplateField),
+			templatesLock:  &sync.RWMutex{},
+			templateTTL:    templateTTL,
+			ianaRegistry:   ianaReg,
+			antreaRegistry: antreaReg,
+			address:        address,
+			maxBufferSize:  maxBufferSize,
+			stopChan:       make(chan bool),
+			packets:        make([]*Packet, 0),
+		},
+		workerList: workerList,
+		workerPool: workerPool,
+		workerNum:  workerNum,
+	}
+	return collectProc, nil
+}
+
+func (cp *collectingProcess) DecodeMessage(msgBuffer *bytes.Buffer) (*Packet, error) {
```

Comment:
DecodePacket better name for function as it essentially decodes packet and returns a message.